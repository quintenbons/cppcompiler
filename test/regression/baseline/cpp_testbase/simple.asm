section .data

section .rodata

section .bss

section .text
	global _start

_start:
	call main
	mov rax, 60                  ; Syscall number for exit (60)
	mov rdi, rbx                 ; Exit code (0) expects return of main to be put in rbx for now
	syscall                      ; Make the syscall

main:
  mov rbx, 1
  ret
	ret
