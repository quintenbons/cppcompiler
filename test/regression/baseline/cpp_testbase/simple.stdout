[35m[Node_TranslationUnit] Function count: 1
[0m[35m  [Node_Function] ReturnType: int ; Name: main ; ParamCount: 2
[0m[35m    [Node_Type] int
[0m[35m    [Node_FunctionParameterList] ParameterCount: 2
[0m[35m      [Node_FunctionParameter] Type: int ; Name: argc
[0m[35m      [Node_FunctionParameter] Type: char** ; Name: argv
[0m[35m    [Node_InstructionList] InstructionCount: 1
[0m[35m      [Node_ReturnStatement] 
[0m[35m      [Node_NumberLiteral] 1
[0m[35m[Node_TranslationUnit] Class count: 0
[0m
== Decorating
[35mScopeStack of 1 scopes
[0m[35m[Scope] id=0 ; parent=0
  [Type] id=1 ; name=int ; size=32
  [Type] id=0 ; name=void ; size=0
[0m
== Done decorating
[35m[Node_TranslationUnit] Function count: 1
[0m[35m  [Node_Function] ReturnType: int ; Name: main ; ParamCount: 2
[0m[35m    [Node_Type] int
[0m[35m    [Node_FunctionParameterList] ParameterCount: 2
[0m[35m      [Node_FunctionParameter] Type: int ; Name: argc
[0m[35m      [Node_FunctionParameter] Type: char** ; Name: argv
[0m[35m    [Node_InstructionList] InstructionCount: 1
[0m[35m      [Node_ReturnStatement] 
[0m[35m      [Node_NumberLiteral] 1
[0m[35m[Node_TranslationUnit] Class count: 0
[0m
== Generating code
== Generated asm to a.asm:
section .data

section .rodata

section .bss

section .text
	global _start

_start:
	call main
	mov rax, 60                  ; Syscall number for exit (60)
	mov rdi, rbx                 ; Exit code (0) expects return of main to be put in rbx for now
	syscall                      ; Make the syscall

main:
	mov rbx, 1
	ret
	ret
== Generated .o as a.o:
== Generating exe as a.out:
